edc-extensions/agreements/retirement-evaluation-store-sql/src/main/java/org/eclipse/tractusx/edc/agreements/retirement/store/SqlAgreementsRetirementStoreExtension.java

```diff
@@ -0,0 +1,76 @@
+/********************************************************************************
+ * Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.tractusx.edc.agreements.retirement.store;
+
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.runtime.metamodel.annotation.Provider;
+import org.eclipse.edc.runtime.metamodel.annotation.Setting;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.sql.QueryExecutor;
+import org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry;
+import org.eclipse.edc.transaction.spi.TransactionContext;
+import org.eclipse.tractusx.edc.agreements.retirement.spi.store.AgreementsRetirementStore;
+import org.eclipse.tractusx.edc.agreements.retirement.store.sql.PostgresAgreementRetirementStatements;
+import org.eclipse.tractusx.edc.agreements.retirement.store.sql.SqlAgreementsRetirementStatements;
+import org.eclipse.tractusx.edc.agreements.retirement.store.sql.SqlAgreementsRetirementStore;
+
+@Extension(value = SqlAgreementsRetirementStoreExtension.NAME)
+public class SqlAgreementsRetirementStoreExtension implements ServiceExtension {
+
+    protected static final String NAME = "SQL Agreement Retirement Store.";
+    private static final String DEFAULT_DATASOURCE_NAME = "agreementretirement";
```

Comment:
line 63 was kept [here](https://github.com/eclipse-tractusx/tractusx-edc/pull/1631/commits/b54f8c3a7cd2e664633b06e778c1cb5d6bf32c09) since is the same approach seen in the other datasources.